Libraries/LibMarkdown/Document.cpp

```diff
@@ -67,33 +67,36 @@ String Document::render_for_terminal() const
 template<typename BlockType>
 static bool helper(Vector<StringView>::ConstIterator& lines, NonnullOwnPtrVector<Block>& blocks)
 {
-    NonnullOwnPtr<BlockType> block = make<BlockType>();
-    bool success = block->parse(lines);
-    if (!success)
+    OwnPtr<BlockType> block = BlockType::parse(lines);
+    if (!block)
         return false;
-    blocks.append(move(block));
+    blocks.append(block.release_nonnull());
     return true;
 }
 
-bool Document::parse(const StringView& str)
+RefPtr<Document> Document::parse(const StringView& str)
```

Comment:
Why is document returned in a `RefPtr` but everything else in a `OwnPtr`?

Not saying it's wrong, but if there's no particular reason for it let's not do it.